sub/sd_ass.c

```diff
@@ -593,6 +593,29 @@ static char *get_text(struct sd *sd, double pts)
     return ctx->last_text;
 }
 
+static struct sd_times get_times(struct sd *sd, double pts)
+{
+    struct sd_ass_priv *ctx = sd->priv;
+    ASS_Track *track = ctx->ass_track;
+    struct sd_times res = { .start = MP_NOPTS_VALUE, .end = MP_NOPTS_VALUE };
+
+    if (pts == MP_NOPTS_VALUE || ctx->duration_unknown)
+        return res;
+
+    long long ipts = find_timestamp(sd, pts);
+
+    for (int i = 0; i < track->n_events; ++i) {
+        ASS_Event *event = track->events + i;
+        if (ipts >= event->Start && ipts < event->Start + event->Duration) {
+            if (res.start == MP_NOPTS_VALUE)
+                res.start = event->Start / (double)1000;
```

Comment:
Does this rely on events being sorted by time? Because that's not the case. If it matters, you maybe have to extend the check.
On a side note, I think "1000.0" is better than "(double)1000".

Reply:
good catch, I indeed assumed they were ordered. fixed (style too) in commits below (meant to be squashed)